pkg/cloudprovider/provider/gce/instance.go

```diff
@@ -56,10 +58,10 @@ func (gi *googleInstance) ID() string {
 }
 
 // Addresses implements instance.Instance.
-func (gi *googleInstance) Addresses() []string {
-	var addrs []string
+func (gi *googleInstance) Addresses() map[string]v1.NodeAddressType {
+	addrs := map[string]v1.NodeAddressType{}
 	for _, ifc := range gi.ci.NetworkInterfaces {
-		addrs = append(addrs, ifc.NetworkIP)
+		addrs[ifc.NetworkIP] = v1.NodeInternalIP
```

Comment:
Doesn't the address type depend on if we assign a public IP address to the instance?

Reply:
I am not quite sure about this since the documentation for this field states clearly that, `NetworkIP` in the `NetworkInterface` is the internal ip Address of the instance. 
https://godoc.org/google.golang.org/api/compute/v1#NetworkInterface